### PR TITLE
Solana Agreement Scaffold

### PR DESCRIPTION
**Pull request by Jaxiii**
#41 Development
    
    "@solana/wallet-adapter-base": "^0.9.23",
    "@solana/wallet-adapter-react": "^0.15.35",
    "@solana/wallet-adapter-react-ui": "^0.9.34",
    "@solana/wallet-adapter-wallets": "^0.19.22",
    "@solana/web3.js": "^1.78.5"
   
  
Added since rainbow do not support the Solana network, just Ethereum/Polygon.
    
Need an ideal UX/UI solution to change between ecosystem, this branch is set to use Solana wallet adapter only, i'm using a lot of hardcoded stuff in the mean time, not ideal.
    
Simple test are passing in Anchor-TS, should be good start from here.
Feedback about the files placements and coding styling would be great.
   
Further contact, please DM me at WEB3DEV Discord, @sedasboy

### CODE REVIEW

**Comment by nomadbitcoin** [comment]
Wow! Thanks for the contribution!

But, in order to accept your pr and be able to work together counting on your expertise in solana to develop Kyodo I need you to make some relevant modifications. 

1 - The application must initialize the contracts according to the selected chain (I'm still implementing this part). But note that the application contexts are already being defined with the help of a ContractManager that has a contract initializer. Currently, the initialization of Ethereum contracts is working perfectly, Solana contracts should be initialized and "distributed" to the application in this way. This way we maintain an agnostic application and can easily add new chains as long as the contracts have the same interface.

2 - Place all the relevant Solana files in the chains/solana folder.

3 - Use selectedChain = solana hardcoded until the chain selection logic is implemented in the AccountContext that will define the account for the entire application. I'm working on it at the moment.

4 - If there are any adaptations that need to be made to any part of the application's architecture to make it work well with solana, please let me know. From now on, I will gradually remove everything that is Ethereum-specific from the other components and make it modularized to be chain agnostic.

File: anchor/programs/agreement/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
+use anchor_lang::prelude::*;
```
Comment (danicuki) [inline anchor/programs/agreement/src/lib.rs:1]: The EVM version of the agreements includes not a single agreement per contract but a list of all agreements created for a single community. The contract has a function `createAgreement` that allows any user to add another agreement to the list. 

In this different architecture proposed here, with a single agreement per account, how easy would be:
 - to list all agreements of a company
 - to list all agreements under a specific community

File: anchor/programs/agreement/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#[account]
+pub struct AgreementAccount {
+    pub id: u128,
+    pub title: String,
+    pub description: String,
+    pub skills: Vec<String>,
+    pub payment_amount: u64,
+    pub professional: Pubkey,
+    pub company: Pubkey,
+    pub token_incentive: Pubkey,
```
Comment (danicuki) [inline anchor/programs/agreement/src/lib.rs:53]: What does it mean this parameter? We know that the community must be part of the agreement. Is this parameter related to the community the professional is connected to?

**Comment by danicuki** [comment]
I am closing this PR, since there are many changes that need to be coded to be accepted. I am more than happy to review a new PR with these modifications, including the #64 features

**Pull request closed without merging**